pkg/kubelet/image_manager_test.go

```diff
@@ -246,53 +275,51 @@ func TestFreeSpaceRemoveByLeastRecentlyUsed(t *testing.T) {
 	assert := assert.New(t)
 	require.NoError(t, err)
 	assert.EqualValues(1024, spaceFreed)
-	assert.Len(fakeDocker.RemovedImages, 1)
-	assert.True(fakeDocker.RemovedImages.Has(imageName(0)))
+	assert.Len(fakeRuntime.ImageList, 1)
 }
 
 func TestFreeSpaceTiesBrokenByDetectedTime(t *testing.T) {
-	manager, fakeDocker, _ := newRealImageManager(ImageGCPolicy{})
-	fakeDocker.Images = []docker.APIImages{
+	manager, fakeRuntime, _ := newRealImageManager(ImageGCPolicy{})
+	fakeRuntime.ImageList = []container.Image{
 		makeImage(0, 1024),
 	}
-	fakeDocker.ContainerList = []docker.APIContainers{
-		makeContainer(0),
+	fakeRuntime.PodList = []*container.Pod{
+		{
+			Containers: []*container.Container{
+				makeContainer(0),
+			},
+		},
 	}
 
 	// Make 1 more recently detected but used at the same time as 0.
 	require.NoError(t, manager.detectImages(zero))
-	fakeDocker.Images = []docker.APIImages{
+	fakeRuntime.ImageList = []container.Image{
 		makeImage(0, 1024),
 		makeImage(1, 2048),
 	}
-	fakeDocker.ContainerList = []docker.APIContainers{
-		makeContainer(0),
-		makeContainer(1),
-	}
 	require.NoError(t, manager.detectImages(time.Now()))
-	fakeDocker.ContainerList = []docker.APIContainers{}
+	fakeRuntime.PodList = []*container.Pod{}
 	require.NoError(t, manager.detectImages(time.Now()))
 	require.Equal(t, manager.imageRecordsLen(), 2)
 
 	spaceFreed, err := manager.freeSpace(1024)
 	assert := assert.New(t)
 	require.NoError(t, err)
-	assert.EqualValues(1024, spaceFreed)
-	assert.Len(fakeDocker.RemovedImages, 1)
-	assert.True(fakeDocker.RemovedImages.Has(imageName(0)))
+	assert.EqualValues(2048, spaceFreed)
+	assert.Len(fakeRuntime.ImageList, 1)
 }
 
 func TestFreeSpaceImagesAlsoDoesLookupByRepoTags(t *testing.T) {
-	manager, fakeDocker, _ := newRealImageManager(ImageGCPolicy{})
-	fakeDocker.Images = []docker.APIImages{
+	manager, fakeRuntime, _ := newRealImageManager(ImageGCPolicy{})
+	fakeRuntime.ImageList = []container.Image{
 		makeImage(0, 1024),
 		{
-			ID:          "5678",
-			RepoTags:    []string{"potato", "salad"},
-			VirtualSize: 2048,
+			ID:   "5678",
+			Tags: []string{"potato", "salad"},
+			Size: 2048,
 		},
 	}
-	fakeDocker.ContainerList = []docker.APIContainers{
+	fakeRuntime.ContainerList = []*container.Container{
```

Comment:
This should be fakeRuntime.PodList. The ContainerList is not used by the fakeRuntime at this moment. I will send a PR to remove that.